### PR TITLE
chore: add release pipeline for the spec

### PR DESCRIPTION
**Pull request by derberg**
This PR introduces release automation for the spec using conventional commits spec and following the release cadence agreement.

assumptions:
- all changes for release happen in release branch and every change in release branch produces a prerelease
- merge from release into master branch triggers normal release, like minor or major
- it is possible to merge from just a fork with for example `fix` PR and trigger patch release

See also: https://github.com/asyncapi/spec/issues/536

---

Commit like `fix: spelling mistake in Operations object description` to master branch

Screenshot 2021-05-19 at 13 43 10

Results in a patch release (base was 2.0.1):
Screenshot 2021-05-19 at 13 45 21

Commit like `feat: add something new in some non-release branch` added to branch that doesn't't follow pattern with `-release` suffx

Screenshot 2021-05-19 at 14 36 21

did not trigger a release

Commit like `feat: add something new in release branch` is a branch that has `2021-06-release` suffix 

Screenshot 2021-05-19 at 15 56 57

Results in proper pre release `v2.1.0-2021-06-release.1`

Screenshot 2021-05-19 at 15 55 19

Another commit to release branch `feat: add something more in release branch` doesn't close bump from 2.1 to 2.2 but proper bump of suffix from 1 to 2

Screenshot 2021-05-19 at 16 02 47

### CODE REVIEW

File: .releaserc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+branches:
+- master
+- name: 2021-06-release
```
Comment (derberg) [inline .releaserc:4]: yes, we need to be hardcoded here 🤷🏼 believe me or not, this is also strange for me but `semantic-release` package doesn't support regex here (I tried everything). I will make sure this step is part of the release process procedure

**Review by magicmatatjahu — APPROVED** [review]

**Review by fmvilas — APPROVED** [review]
I like it. It's exactly what I'd do manually but automated 💯

**Comment by derberg** [comment]
It is actually not included in the release. This was an unintentional release caused by the release automation initial run. 1.0 version of the spec was released a long time ago. The release mentioned by the bot was removed.

**Comment by derberg** [comment]
Not included in release 2.1 yet, 2.1 was mistakenly triggered by our automation.

**Pull request merged**